File: problems/trees/leafList.js
const leafList = (root) => {
  // todo
  if(!root) return []
  let listy = []
  getLeafs(root, listy)
  
  return listy 
};

const getLeafs = (root, list) => {
  if(!root.left && !root.right) list.push(root.val)
  
  if(root.left) getLeafs(root.left, list)
  if(root.right) getLeafs(root.right, list)

}

File: python_Structy/leaf_list.py
# class Node:
#   def __init__(self, val):
#     self.val = val
#     self.left = None
#     self.right = None

def leaf_list(root):
  if root is None: return []
  if root.left is None and root.right is None: return [root.val] 


  left_list = leaf_list(root.left)
  right_list = leaf_list(root.right)

  return [*left_list, *right_list]